src/sethostid.rs/src/main.rs

```diff
@@ -0,0 +1,82 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2021 Joyent, Inc.
+ */
+
+/*
```

Comment:
I did bring it up in chat a few weeks ago and he and I talked about it a bit. He helped me decide on ultimately shipping the rust one instead of the C one.

But also, additional scrutiny is welcome :-)

Reply:
My Rust knowledge is minimal (haven't sat down and learned the damned language yet), so I hoped he'd take a peek.  I've no issue with Rust per se.